prqlc/prql-compiler/src/ir/pl/fold.rs

```diff
@@ -74,8 +74,8 @@ pub fn fold_expr_kind<T: ?Sized + PlFold>(fold: &mut T, expr_kind: ExprKind) ->
     Ok(match expr_kind {
         Ident(ident) => Ident(ident),
         All { within, except } => All {
-            within,
-            except: fold.fold_exprs(except)?,
+            within: Box::new(fold.fold_expr(*within)?),
+            except: Box::new(fold.fold_expr(*except)?),
```

Comment:
Quick Q — are these boxing to try and fix the stack issue? Or just helpful per se?

Reply:
Not exactly - I've changed `within: Ident` into `within: Expr` and `except: Vec<Expr>` into `except: Expr`. I have a general rule that every reference to Expr in AST should be either a `Vec` or a `Box`, because `Expr` is really big and without `Box`, `ExprKind` would have to grow substantially.

Reply:
Perfect, that makes sense